### ISSUE TITLE
[eCBIS] Restrict clients who can be enrolled to FP

### ISSUE THREAD

**Issue by AnnieMungai** [opensrp/fhircore#1601]
**Describe the bug**
v.0.1.2 - preview

Currently, on the app - all clients are able to access FP enrollment meant to form. This form should be accessible only to clients of reproductive age - 15- 49yrs., who are not enrolled to an ANC careplan. 

**To Reproduce**
1. Register a client at any age. 
2. Navigate to their profile.
3. Open the top right-hand menu. See the option to Add to family planning. 
4. Tap on this option - the form opens. 

**Required**

- [ ] Make the FP registration form only accessible to persons between 15-49yrs. 
- [ ] FP form available for clients who are not ANC clients. 

**Acceptance criteria**

- The Add to family planning option should only be available to clients who are 15-49yrs old. 
- Register family planning should not be available to any ANC client / client marked to be pregnant.

**Comment by pld** [comment]
Just checking, the requirement is that the be 15-49yrs AND not an ANC client, is that correct?

**Comment by AnnieMungai** [comment]
@pld yes. That is correct.

**Comment by pld** [comment]
Cool thanks, I'm not sure if this is a bug or a feature, I haven't been in discussion of it before, but maybe have just missed it.

@dubdabasoduba is this in the scope of what the profile config is built to manage?

**Comment by dubdabasoduba** [comment]
@pld Yes, this rules engine will be able to check the presence of the Condition that marks FP patients.

**Comment by dubdabasoduba** [comment]
The idea with the Overflow Profile menus is to have visibility defined by rules engines.

**Comment by pld** [comment]
Cool, thanks, do we an in progress issue related to this (which can close this)? Or do we have an error in the configs for this app (then this can be closed w/a new fhir-resources issue to fix that)? Or is this already implemented and there's an error (we keep this open and assign it)?

**Comment by f-odhiambo** [comment]
Hey @AnnieMungai why did we mark this as `Blocked`?

**Comment by AnnieMungai** [comment]
@f-odhiambo it was dependent on the some other work that was not yet completed on fhircore. cc @dubdabasoduba
